search/api.py

```diff
@@ -0,0 +1,131 @@
+"""
+Functions for search functionality.
+"""
+
+from __future__ import unicode_literals
+
+from haystack.inputs import Exact
+from haystack.query import SearchQuerySet
+
+from search.sorting import LoreSortingFields
+from taxonomy.models import Vocabulary
+from ui.views import get_vocabularies
+
+
+def construct_queryset(repo_slug, query='', selected_facets=None, sortby=''):
+    """
+    Create a SearchQuerySet given search parameters.
+
+    Args:
+        repo_slug (learningresources.models.Repository):
+            Slug for repository being searched.
+        query (unicode): If present, search phrase to use in queryset.
+        selected_facets (list or None):
+            If present, a list of facets to narrow the search with.
+        sortby (unicode): If present, order by this sorting option.
+    Returns:
+        haystack.query.SearchQuerySet: The queryset.
+    """
+    if selected_facets is None:
+        selected_facets = []
+
+    queryset = SearchQuerySet()
+
+    kwargs = {}
+    for facet in selected_facets:
+        queryset = queryset.narrow(facet)
+
+    if query != "":
+        kwargs["content"] = query
+
+    queryset = queryset.filter(**kwargs)
+
+    queryset = queryset.filter(repository=Exact(repo_slug))
+
+    if sortby == "":
+        sortby = LoreSortingFields.DEFAULT_SORTING_FIELD
```

Comment:
nit: you can write something like
`sortby = sortby or LoreSortingFields.DEFAULT_SORTING_FIELD`

(I'm pretty sure I wrote that line originally, so blame on me :-) )